packages/components/src/internal/components/samples/SampleAssayDetail.tsx

```diff
@@ -78,38 +100,115 @@ const SampleAssayDetailButtons: FC<SampleAssayDetailButtonsProps> = props => {
     }
 };
 
+const SampleAssayDetailButtonsRight: FC<SampleAssayDetailButtonsProps> = props => {
+    const { activeSampleAliquotType, onSampleAliquotTypeChange, isSourceSampleAssayGrid } = props;
+
+    return (
+        <>
+            <SampleAliquotViewSelector
+                aliquotFilterMode={activeSampleAliquotType}
+                updateAliquotFilter={onSampleAliquotTypeChange}
+                headerLabel={
+                    isSourceSampleAssayGrid ? 'Show Assay Data with Source Samples' : 'Show Assay Data with Samples'
+                }
+                samplesLabel={isSourceSampleAssayGrid ? 'Derived Samples Only' : 'Sample Only'}
+                allLabel={isSourceSampleAssayGrid ? 'Derived Samples or Aliquots' : 'Sample or Aliquots'}
+            />
+        </>
+    );
+};
+
 interface OwnProps {
     tabOrder: string[];
+    onSampleAliquotTypeChange?: (mode: ALIQUOT_FILTER_MODE) => any;
+    activeSampleAliquotType?: ALIQUOT_FILTER_MODE;
+    showImportBtn?: boolean;
+    isSourceSampleAssayGrid?: boolean;
+    onTabChange: (tabId: string) => any;
+    activeTabId?: string
 }
 
 type SampleAssayDetailBodyProps = Props & InjectedAssayModel & OwnProps;
 
 const SampleAssayDetailBodyImpl: FC<SampleAssayDetailBodyProps & InjectedQueryModels> = memo(props => {
-    const { actions, assayModel, queryModels, sampleModel, tabOrder } = props;
+    const {
+        actions,
+        assayModel,
+        queryModels,
+        sampleModel,
+        tabOrder,
+        showImportBtn,
+        isSourceSampleAssayGrid,
+        showAliquotViewSelector,
+        onSampleAliquotTypeChange,
+        activeSampleAliquotType,
+        emptyAssayDefDisplay,
+        emptyAssayResultDisplay,
+        emptyAliquotViewMsg,
+        emptySampleViewMsg,
+        onTabChange,
+        activeTabId
+    } = props;
     const allModels = Object.values(queryModels);
     const allLoaded = allModels.every(model => !model.isLoading);
 
     useEffect(() => {
         actions.loadAllModels(true);
     }, []);
 
+    // if (showAliquotViewSelector && activeSampleAliquotType != ALIQUOT_FILTER_MODE.all)
     const { queryModelsWithData, tabOrderWithData } = useMemo(() => {
         const models = {};
         const tabOrderWithData = tabOrder.slice();
-        Object.values(queryModels).forEach(model => {
+        let targetQueryModels = Object.values(queryModels);
+        const isFilteredView =
+            showAliquotViewSelector &&
+            activeSampleAliquotType != null &&
+            activeSampleAliquotType != ALIQUOT_FILTER_MODE.all;
+        if (isFilteredView) {
+            targetQueryModels = [];
+            Object.values(queryModels).forEach(model => {
+                if (model.id?.indexOf('unfiltered-assay-detail') === 0) targetQueryModels.push(model);
```

Comment:
Having the ids carry some meaning is perhaps a little fragile.  I think we should at least create constants for the expected parts of the ids.